### PR TITLE
Ability to save multi-animal pose tracks to single-animal files

### PR DESCRIPTION
**Pull request by DhruvSkyy**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**What does this PR do?**

This PR adds the functionality to save xarrays as a dictionary of individual pandas dataframes for to_dlc_df and as multiple files for each individual for to_dlc_file. This is based on the split_individuals: bool parameter, if True it will split the data for each individual, if false it will save it as a multi-animal file. 

## References

This PR addresses Issue #39. 

## How has this PR been tested?

The PR was tested on a Juypter Notebook against the four scenarios, (credit to @niksirbi):
* The xarray.Dataset has multiple (>1) individuals and the user wants to save it to a multi-animal DeepLabCut dataframe. This currently works with the existing functions.
* The xarray.Dataset has only 1 individual but the user still wants to save it to a multi-animal DeepLabCut dataframe, including the redundant "individuals" level. This also works currently.
* The xarray.Dataset has a single individual and the user wants to save it to single-animal DeepLabCut dataframe (without the "individuals" level). This behaviour needs to be implemented, the output will be a single file.
* The xarray.Dataset has multiple (>1) individuals and the user wants to split them for saving into multiple single-animal files. * This is the trickiest case to implement, as the dataset would have to be split across individuals (behind the scenes) and each part is then saved to a single-animal file as in case 3. The output should be as many files as there are individuals.

## Todo

Need to write tests. 

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: movement/io/save_poses.py [hunk 1]
```diff
@@ -87,11 +186,32 @@ def to_dlc_file(ds: xr.Dataset, file_path: Union[str, Path]) -> None:
     file_path : pathlib Path or str
         Path to the file to save the DLC poses to. The file extension
         must be either ".h5" (recommended) or ".csv".
+    split_individuals : bool, optional
+        Format of the DeepLabcut output file.
```
Comment (niksirbi) [inline movement/io/save_poses.py:190]: I would remove this line. Also, I discovered that "-" dashes do not rendr as bullet points in Sphinx (so they won't appear correctly in our API docs). It's ok to put the whole description of "split_individuals" in a single paragraph.

File: movement/io/save_poses.py [hunk 2]
```diff
@@ -87,11 +186,32 @@ def to_dlc_file(ds: xr.Dataset, file_path: Union[str, Path]) -> None:
...
+        individuals in the dataset: True if there is only one, and
+        False if there are more than one. This is the default.
 
     See Also
     --------
     to_dlc_df : Convert an xarray dataset containing pose tracks into a
-        DeepLabCut-style pandas DataFrame with multi-index columns.
+    DeepLabCut-style pandas DataFrame with multi-index columns
+    for each individual or a dictionary of DataFrames for each individual
+    based on the 'split_individuals' argument.
```
Comment (niksirbi) [inline movement/io/save_poses.py:208]: For this docstring to render correctly in Sphinx, these lines need to be indented:

```suggestion
        DeepLabCut-style pandas DataFrame with multi-index columns
        for each individual or a dictionary of DataFrames for each individual
        based on the 'split_individuals' argument.
```

**Review by niksirbi — CHANGES_REQUESTED** [review]
Very well done @DruvSkyy :tada:

I like the current implementation of this feature. Thanks to our back-and-forth, we converged on something sensible I think.

I have left some minor comments for you to implement.

After that, I'll modify the tests to cover the new funcitonality. Let's see if they'll help us discover any bugs/edge cases.

File: movement/io/save_poses.py [hunk 3]
```diff
@@ -11,72 +11,171 @@
 logger = logging.getLogger(__name__)
 
 
-def to_dlc_df(ds: xr.Dataset) -> pd.DataFrame:
-    """Convert an xarray dataset containing pose tracks into a
-    DeepLabCut-style pandas DataFrame with multi-index columns.
+def _xarraytodf(ds: xr.Dataset, columns: pd.MultiIndex) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply movement/io/save_poses.py]: sorry, my bad, I had a typo in there. I've renamed it to `_xarray_to_dlc_df`

**Comment by niksirbi** [comment]
Hey @DhruvSkyy, I've made the following modification to this PR:
- I've reviewed all docstring and function/argument names in `save_poses.py` and made sure they are consistent throughout
- I've fixed the tests that were failing 
- I've added new tests for the feature you've implemented. You can find these on `tests/test_unit/test_save_poses.py`, specifically:
	- `test_auto_split_individuals`
	- `test_to_dlc_df_split_individuals`
	- `test_to_dlc_file_split_individuals`
	
Take a look at the new tests and see if you can make sense of them. I'm happy to discuss these 1-to-1 if some things are unclear.

Other than that, this PR is now ready to merge I think, thanks a lot for your help!

**Review by niksirbi — APPROVED** [review]
